File: khubdeemju/src/main/java/com/itsci/khubdeemju/model/Student.java
package com.itsci.khubdeemju.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Student")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Student {

    @Id
    @Column(name = "student_id", length = 10)
    private String student_id;
    @Column(name = "firstname", length = 50, nullable = false)
    private String firstname;
    @Column(name = "lastname", length = 50, nullable = false)
    private String lastname;
    @Column(name = "major", length = 40, nullable = false)
    private String major;
    @Column(name = "faculty", length = 40, nullable = false)
    private String faculty;
    @Column(name = "student_score", nullable = false)
    private int student_score;
    @Column(name = "img_student")
    private String img_student;
    @Column(name = "email", length = 50, nullable = false)
    private String email;
    @Column(name = "password", length = 20, nullable = false)
    private String password;

}

File: Project_khubdee_mju_application/lib/model/student.dart
class Student {
  String? student_id;
  String? firstname;
  String? lastname;
  int? student_score;
  String? major;
  String? faculty;
  String? img_student;
  String? email;
  String? password;

  Student(
      {this.student_id,
      this.firstname,
      this.lastname,
      this.student_score,
      this.major,
      this.faculty,
      this.img_student,
      this.email,
      this.password});

  Map<String, dynamic> fromStudentToJson() {
    return <String, dynamic>{
      'student_id': student_id,
      'firstname': firstname,
      'lastname': lastname,
      'student_score': student_score,
      'major': major,
      'faculty': faculty,
      'img_student': img_student,
      'email': email,
      'password': password,
    };
  }

  factory Student.fromJsonToStudent(Map<String, dynamic> json) => Student(
      student_id: json['student_id'],
      firstname: json['firstname'],
      lastname: json['lastname'],
      student_score: json['student_score'],
      major: json['major'],
      faculty: json['faculty'],
      img_student: json['img_student'],
      email: json['email'],
      password: json['password']);
}
